be/src/storage/lake/compaction_policy.cpp

```diff
@@ -115,7 +115,10 @@ StatusOr<std::vector<RowsetPtr>> BaseAndCumulativeCompactionPolicy::pick_cumulat
     std::vector<RowsetPtr> input_rowsets;
     uint32_t cumulative_point = _tablet_metadata->cumulative_point();
     uint32_t segment_num_score = 0;
+    bool restrict_segment_count = config::enable_compaction_strict_segment_count_check;
```

Comment:
suggest that old cumulative compaction not do partial segments compaction, because old cumulative compaction will update the cumulative point to the new output rowset which is non-overlap.
the old cumulative compaction may be removed later.

Reply:
done